ssl/record/methods/tls_common.c

```diff
@@ -2129,7 +2137,10 @@ int tls_free_buffers(OSSL_RECORD_LAYER *rl)
     /* Read direction */
 
     /* If we have pending data to be read then fail */
-    if (rl->curr_rec < rl->num_recs || TLS_BUFFER_get_left(&rl->rbuf) != 0)
+    if (rl->curr_rec < rl->num_recs
+            || rl->curr_rec != rl->num_released
```

Comment:
I have question: is `rl->curr_rec != rl->num_released` the right cmp operation here? The reason I'm asking is that elsewhere in this file we use a `<` comparison. For example here `tls_release_record()`:
```
1155 int tls_release_record(OSSL_RECORD_LAYER *rl, void *rechandle, size_t length)
1156 {
1157     TLS_RL_RECORD *rec = &rl->rrec[rl->num_released];
1158
1159     if (!ossl_assert(rl->num_released < rl->curr_rec)
1160             || !ossl_assert(rechandle == rec)) {
1161         /* Should not happen */
```
perhaps comparison type does not matter in this case. I'm rather curious if there is a reason for inconsistency.

Reply:
`rl->num_released` should always be <= `rl->curr_rec`.

In the case of the code that you reference we are asserting the stricter condition that `rl->num_released < rl->curr_rec` since it would be a libssl bug to call `tls_release_record` where `rl->num_released >= rl->curr_rec`.

In this case it is the other way around. This function must fail if `rl->curr_rec != rl->num_released`. It would be sufficient to fail if `rl->num_released < rl->curr_rec` if we rely on the fact that `rl->num_released` should never be greater than `rl->curr_rec` - but I prefer the stricter check here.